src/newrelic_telemetry_sdk/batch.py

```diff
@@ -56,3 +58,41 @@ def flush(self):
 
         common = self._common and self._common.copy()
         return batch, common
+
+
+class SpanBatch(Batch):
+    """Aggregates spans, providing a record / flush interface.
+
+    :param tags: (optional) A dictionary of tags to attach to all flushes.
+    :type tags: dict
+    """
+
+
+class EventBatch(Batch):
+    """Aggregates events, providing a record / flush interface.
+
+    :param tags: (optional) A dictionary of tags to attach to all flushes.
+    :type tags: dict
+    """
+
+    def __init__(self, tags=None):
+        super(EventBatch, self).__init__(tags)
+        if tags:
+            # Events are currently a flat structure, so there's no nesting of
+            # user attributes.
+            self._common = dict(tags)
+
+    def flush(self):
+        """Flush all items from the batch
+
+        This method returns all items in the batch.
+
+        The batch is cleared as part of this operation.
+
+        :returns: A tuple of (items,)
+        :rtype: tuple
+        """
+        items, common = super(EventBatch, self).flush()
+        if common:
+            items = tuple(dict(common, **item) for item in items)
```

Comment:
I had to look up how this worked to verify the behavior, when you initialize a dictionary with a dictionary and kwargs, it will copy the dictionary and update the keys with the kwargs. So if there are keys in common that are duplicated by in the event the value from the event will be used.

Reply:
Yup! There's a test for this as well!
I've been thinking - should we be adding this common tags feature right now?